ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/cache/LRUCache.java

```diff
@@ -42,12 +43,12 @@ public LRUCache(int capacity) {
 
   private LRUCache(int capacity, Map<K, V> initialCachedContent) {
     this.maxCapacity = capacity;
-    LimitedHashMap<K, V> cacheMap = new LimitedHashMap<>(maxCapacity);
+    Map<K, V> cacheMap = LimitedMap.create(maxCapacity, LimitStrategy.DROP_LEAST_RECENTLY_ACCESSED);
     // copy safely, initialCachedContent is always a SynchronizedMap instance
     synchronized (initialCachedContent) {
```

Comment:
I agree with @ajsutton that synchronizing a collection would better to perform on a client side. This place is a good illustration: we need to make sure that the collection is `synchronized` here, but `LimitedMap.create` doesn't imply returning synchronized collection. 
Alternative option would be to have two factory variants like: 
- `LimitedMap.create`
- `LimitedMap.createSynchronized`

Reply:
Good points!  I split up the helpers so we now have:
- `LimitedMap.create`
- `ConcurrentLimitedMap.create`
- `LimitedSet.create`
- `ConcurrentLimitedSet.create`